.github/workflows/nextjs-integration-test.yml

```diff
@@ -0,0 +1,220 @@
+# Reusable workflow to execute certain version of Next.js integration tests
+# with turbopack.
+#
+# Refer test.yml for how this workflow is being initialized
+# - Workflow can specify `inputs.version` to specify which version of next.js to use, otherwise will use latest release version.
+name: Turbopack Next.js integration test
+
+on:
+  workflow_call:
+    inputs:
+      # Allow to specify Next.js version to run integration test against.
+      # If not specified, will use latest release version including canary.
+      version:
+        type: string
+      # The base of the test results to compare against. If not specified, will try to compare with latest main branch's test results.
+      diff_base:
+        type: string
+        default: 'main'
+      force_post_to_slack:
+        type: boolean
+      # Skip posting to slack regardless of the conditions.
+      skip_post_to_slack:
+        type: boolean
+        default: false
+
+# Workflow-common env variables
+env:
+  # Enabling backtrace will makes snapshot tests fail
+  RUST_BACKTRACE: 0
+  NEXT_TELEMETRY_DISABLED: 1
+  TEST_CONCURRENCY: 6
+  DATADOG_API_KEY: ${{ secrets.DATA_DOG_API_KEY }}
+  DATADOG_TRACE_NEXTJS_TEST: 'true'
+  DD_ENV: 'ci'
+  # Turbopack specific customization for the test runner
+  TURBOPACK: 1
+  __INTERNAL_CUSTOM_TURBOPACK_BINDINGS: ${{ github.workspace }}/packages/next-swc/native/next-swc.linux-x64-gnu.node
+  NEXT_TEST_SKIP_RETRY_MANIFEST: ${{ github.workspace }}/integration-test-data/test-results/main/failed-test-path-list.json
+  NEXT_TEST_CONTINUE_ON_ERROR: TRUE
+  NEXT_E2E_TEST_TIMEOUT: 240000
+  NEXT_TEST_JOB: 1
+
+jobs:
+  # First, build next-dev and Next.js both to execute across tests.
+  setup_nextjs:
+    name: Setup Next.js build
+    uses: ./.github/workflows/setup-nextjs-build.yml
```

Comment:
Doesn't look like this exists in this repo, checking to see if we even need it...

Reply:
this probably should use next.js's existing workflow to setup next.js build.

Reply:
hmm looks like there's some other jobs/handling in that workflow so copied it over for now

Reply:
I mean eventually for the followup, not immediately in this PR